### PR TITLE
[RyuJIT/ARM32] Fix assertion in 'unreached' when using emitIns_R_L

### PR DESCRIPTION
**Pull request by sjsinju**
Now, All of CodeGenBringUpTests have failed.

message : Assertion failed 'unreached' in 'DomainNeutralILStubClass:IL_STUB_PInvoke(ref)'

I found the problem was caused by #11377 that 'genCodeForTreeNode' is merged to codegenarmarch.cpp.

### CODE REVIEW

File: src/jit/codegenarmarch.cpp [hunk 1]
```diff
@@ -343,7 +343,11 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
         case GT_LABEL:
             genPendingCallLabel       = genCreateTempLabel();
             treeNode->gtLabel.gtLabBB = genPendingCallLabel;
+#if defined(_TARGET_ARM)
```
Comment (mskvortsov) [inline src/jit/codegenarmarch.cpp]: `_TARGET_ARM_` as well as `_TARGET_ARM64_` should both have an ending underscore.

File: src/jit/codegenarmarch.cpp [hunk 2]
```diff
@@ -343,7 +343,11 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
...
+            emit->emitIns_J_R(INS_adr, EA_PTRSIZE, genPendingCallLabel, treeNode->gtRegNum);
```
Comment (mskvortsov) [inline src/jit/codegenarmarch.cpp]: Why not simply `targetReg` instead of `treeNode->gtRegNum`?

Comment (mskvortsov) [inline src/jit/codegenarmarch.cpp]: And generally speaking, as this usage of `emitIns_J_R()` was copied from `codegenlegacy.cpp`, may it be worth to refactor arm32 `emitIns_R_L()` instead? Though I'm not sure what is the difference intended between the `J_R` and `R_L` forms.

File: src/jit/codegenarmarch.cpp (hunk 1 above)
Comment (sjsinju) [inline src/jit/codegenarmarch.cpp]: Thank you. That's my mistake.

File: src/jit/codegenarmarch.cpp (hunk 2 above)
Comment (sjsinju) [inline src/jit/codegenarmarch.cpp]: @mskvortsov  I don't know deeply. But I just copied that from the original source of codegenarm.cpp. And then, The assertion has resolved. 
It seems to need more investigate deeply like what you said for using emitIns_R_L insted of emitIns_J_R().

**Comment by BruceForstall** [comment]
Sorry about that. We should really fix arm32/arm64 to use the same function name in this case (probably _R_L)

**Pull request merged**